daemon/algod/api/server/v2/dryrun.go

```diff
@@ -0,0 +1,512 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package v2
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// DryrunRequest object uploaded to /v2/teal/dryrun
+// It is the same as generated.DryrunRequest but Txns deserialized properly.
+// Given the Transactions and simulated ledger state upload, run TEAL scripts and return debugging information.
+// This is also used for msgp-decoding
+type DryrunRequest struct {
+	// Txns is transactions to simulate
+	Txns []transactions.SignedTxn `codec:"txns"` // not supposed to be serialized
+
+	// Optional, useful for testing Application Call txns.
+	Accounts []generated.Account `codec:"accounts"`
```

Comment:
could you limit size using allocbound ?

Reply:
@algorandskiy should this directory be included in `MSGP_GENERATE` in the `Makefile`?

Reply:
I'm was trying to avoid adding these api struct to msgp, so it is only serialized in `libgoal.go:MakeDryrunStateBytes` with `protocol.EncodeReflect`